assets/js/editor-components/search-list-control/utils.tsx

```diff
@@ -77,7 +77,7 @@ export const buildTermsTree = (
 	delete termsByParent[ '0' ];
 
 	// anything left in termsByParent has no visible parent
-	forEach( termsByParent, ( terms ) => {
+	Object.values( termsByParent ).forEach( ( terms ) => {
```

Comment:
Had a quick look at this one before and unfortunately the forEach from lodash works a bit different than  native array forEach on `Object.values`. This will have unit tests failing.

This is because we rely here on the fact that lodash operates on a mutable `termsByParent` object that changes as `fillWithChildren` is called and when we use `Object.values( termsByParent ).forEach` it ignores mutation of `termsByParent` and iterates over sealed a new enumerable it gets from Object.values. 

This causes duplicate/false values to be added to the tree.

Reply:
O I didn't see duplication testing locally. Ill take another look thanks.

Reply:
I pushed a change which made tests pass. Does that code make more sense to you?

Reply:
looks good! I think it's clearer what (and how) it does it. Was super surprised when I've noticed that it relies on the collection being mutated. Had a look at how it's implemented in lodash and... was quite a trip, also they didn't aim for "simple" :)